admiral/pkg/clusters/handler.go

```diff
@@ -58,43 +61,65 @@ func handleDependencyRecord(sourceIdentity string, r *RemoteRegistry, rcs map[st
 
 		//for every cluster the source identity is running, add their istio ingress as service entry address
 		tempDeployment := rc.DeploymentController.Cache.Get(sourceIdentity)
-		if tempDeployment != nil {
+		//If there is  no deployment, check if a rollout is available.
+		tempRollout := rc.RolloutController.Cache.Get(sourceIdentity)
+		if tempDeployment != nil || tempRollout !=nil {
 			sourceClusters[rc.ClusterID] = rc.ClusterID
 			r.AdmiralCache.IdentityClusterCache.Put(sourceIdentity, rc.ClusterID, rc.ClusterID)
 		}
 
+
+
 		//create and store destination service entries
 		for _, destinationCluster := range destinationIdentitys {
 			destDeployment := rc.DeploymentController.Cache.Get(destinationCluster)
-			if destDeployment == nil {
-				continue
-			}
+			destRollout := rc.RolloutController.Cache.Get(destinationCluster)
+			var tmpSe *networking.ServiceEntry
 
-			//deployment can be in multiple clusters, create SEs for all clusters
+             // Assumption :- We will have either a deployment or  a rollout mapped to an identity but never both in a cluster
+			if destDeployment != nil {
+				//deployment can be in multiple clusters, create SEs for all clusters
 
-			for _, deployment := range destDeployment.Deployments {
+				for _, deployment := range destDeployment.Deployments {
 
-				r.AdmiralCache.IdentityClusterCache.Put(destinationCluster, rc.ClusterID, rc.ClusterID)
+					r.AdmiralCache.IdentityClusterCache.Put(destinationCluster, rc.ClusterID, rc.ClusterID)
 
-				deployments := rc.DeploymentController.Cache.Get(destinationCluster)
+					deployments := rc.DeploymentController.Cache.Get(destinationCluster)
```

Comment:
This may be unrealistic... but it seems to me like it may make sense to create a new interface for `deployment-like` types (so far just rollouts and deployments, but that would make it more extensible). 

Then we could keep some sort of ordered list of implementations (ordered by precedence priority), and iterate through the same logic for each, instead of this if-else block.

In pseudocode something like:
```
for each DeploymentLikeInterface controller in DeploymentLikeInterfaces:
    get()
    doCacheOperations()
    createServiceEntry()
```

Just a thought - let me know if you think that's unfeasible/a bad idea.